global/concourse/templates/daemonset.yaml

```diff
@@ -0,0 +1,139 @@
+  labels:
+    app: {{ $.Values.worker.name }}-{{ .name }}
+    chart: "{{ $.Chart.Name }}-{{ $.Chart.Version }}"
+    release: "{{ $.Release.Name }}"
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: {{ $.Values.worker.name }}-{{ .name }}
+  template:
+    metadata:
+      labels:
+        app: {{ $.Values.worker.name }}-{{ .name }}
+        release: "{{ $.Release.Name }}"
+    spec:
+      nodeSelector:
+        ccloud.sap.com/nodepool: {{ .nodePool | quote }}
+      tolerations:
+      - key: "team"
+        operator: Equal
+        value: {{ .name | quote }}
+        effect: "NoSchedule"
+      serviceAccountName: concourse-worker
+      initContainers:
+      - name: {{ $.Values.worker.name }}-{{ .name }}-init-rm
+        image: "{{ $.Values.concourse.image }}:{{ $.Values.concourse.imageTag }}"
+        imagePullPolicy: {{ $.Values.concourse.imagePullPolicy | quote }}
+        command:
+        - /bin/sh
+        args:
+        - -ce
+        - |-
+          rm -rf {{ $.Values.concourse.concourse.worker.workDir }}/*
+        volumeMounts:
+        - name: concourse-work-dir
+          mountPath: {{ $.Values.concourse.concourse.worker.workDir | quote }}
+      containers:
+      - name: {{ $.Values.worker.name }}-{{ .name }}
+        image: "{{ $.Values.concourse.image }}:{{ $.Values.concourse.imageTag }}"
+        imagePullPolicy: {{ $.Values.concourse.imagePullPolicy | quote }}
+        command:
+        - /bin/sh
+        args:
+        - -ce
+        - |-
+          update-ca-certificates &&
+          /usr/local/concourse/bin/concourse worker
+        livenessProbe:
+          failureThreshold: 5
+          initialDelaySeconds: 10
+          periodSeconds: 15
+          timeoutSeconds: 3
+          httpGet:
+            path: /
+            port: 8888
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - "kill"
+              - "-s"
+              - "SIGUSR2"
+              - "1"
+        env:
+        {{- if ne .name "main" }}
+        - name: CONCOURSE_TEAM
+          value: {{ .name | quote }}
+        {{- end }}
+        - name: CONCOURSE_BIND_DEBUG_PORT
+          value: "9099"
+        - name: CONCOURSE_WORK_DIR
+          value: {{ $.Values.concourse.concourse.worker.workDir | quote }}
+        - name: CONCOURSE_BIND_IP
+          value: "127.0.0.1"
+        - name: CONCOURSE_BIND_PORT
+          value: "7777"
+        - name: CONCOURSE_LOG_LEVEL
+          value: "info"
+        - name: CONCOURSE_TSA_HOST
+          value: "concourse-web:2222"
+        - name: CONCOURSE_TSA_PUBLIC_KEY
+          value: "{{ $.Values.concourse.worker.keySecretsPath }}/host_key.pub"
+        - name: CONCOURSE_TSA_WORKER_PRIVATE_KEY
+          value: "{{ $.Values.concourse.worker.keySecretsPath }}/worker_key"
+        - name: CONCOURSE_GARDEN_BIND_PORT
+          value: "7777"
+        - name: CONCOURSE_BAGGAGECLAIM_DRIVER
+          value: "naive"
```

Comment:
This is pretty bad performance wise, we need to use the standard btrfs one. I guess that's a little tricky with the Kubernetes volumes but this won't scale out all. Ideally the volume should be formatted with btrfs.